obspy/io/mseed/tests/test_mseed_reading_and_writing.py

```diff
@@ -155,6 +155,46 @@ def test_read_and_write_traces(self):
                     np.testing.assert_array_equal(this_stream[0].data,
                                                   new_stream[0].data)
 
+
+    def test_downcast_int64_to_int32(self):
+        """
+        Checks that sample stream of dtype int64 can be downcast to int 32 and written to mseed
+        by downcasting the data to int32 type data. 
+        """
+        # create a dummy stream with int64 data 
+        x = np.array([1,2,-3,4], dtype = np.int64)
+        tr = Trace(x)
+        st = Stream()
+        st.append(tr)
+
+        # make sure the data can be written to mseed -> write to io.BytesIO() object in test 
+        with io.BytesIO() as buf:
+            st.write(buf, format="mseed")
```

Comment:
Please also add a check that the original data in the traces is still of dtype `int64`.

Reply:
Do you mean after st.write() add a check that st is still of dtype int64?

Reply:
exactly